src/lib/utils/SynapseClient.ts

```diff
@@ -1585,6 +1606,42 @@ export const getFileHandleContent = (
   })
 }
 
+export const getFileResult = (
+  fileEntity: FileEntity,
+  sessionToken?: string,
+  includeFileHandles?: boolean,
+  includePreSignedURLs?: boolean,
+  includePreviewPreSignedURLs?: boolean,
+): Promise<FileResult> => {
+
+  return new Promise((resolve, reject) => {
+    const fileHandleAssociationList: FileHandleAssociation[] = [
```

Comment:
This is an array of FileHandleAssociations.  Currently you're creating one, but this could ask for all of them in a single service call.

Reply:
Actually this function is only used for retrieving the download link now after code refactoring. So I only need 1 file handle data to get the download link when a user click the download button instead of a list. Is there a function in SynapseClient I can re-use to just get just 1 file handle data?

Reply:
Sorry, I was looking at old code.  This is great.

Reply:
May want to change other functions in SynapseClient to use this getFileResult()!  I'm looking at getFileEntityFileHandle() and getFileEntityContent(),